site/src/pages/WorkspacePage/WildcardHostnameWarning.tsx

```diff
@@ -0,0 +1,44 @@
+import AlertTitle from "@mui/material/AlertTitle";
+import type { WorkspaceAgent } from "api/typesGenerated";
+import { Alert, AlertDetail } from "components/Alert/Alert";
+import { Link } from "components/Link/Link";
+import { useProxy } from "contexts/ProxyContext";
+import type { FC } from "react";
+import { docs } from "utils/docs";
+
+interface WildcardHostnameWarningProps {
+	agent: WorkspaceAgent;
+}
+
+export const WildcardHostnameWarning: FC<WildcardHostnameWarningProps> = ({
+	agent,
+}) => {
+	const { proxy } = useProxy();
+
+	if (proxy.proxy?.wildcard_hostname) {
+		return null;
+	}
+
+	const hasSubdomainApps = agent.apps?.some((app) => app.subdomain);
+	if (!hasSubdomainApps) {
+		return null;
+	}
+
+	return (
+		<Alert severity="warning">
+			<AlertTitle>Some workspace applications will not work</AlertTitle>
+			<AlertDetail>
+				<div>
+					One or more apps in this workspace have subdomain = true, which
+					requires a Coder deployment with a Wildcard Access URL configured.
+					Please contact your administrator.
```

Comment:
```suggestion
					One or more apps in this workspace have `subdomain = true`, which
					requires a Coder deployment with a Wildcard Access URL configured.
					Please contact your administrator.
```
Not sure if we support markdown here, but it would be nice.

Reply:
Done